internal/worker_version_sets.go

```diff
@@ -130,6 +155,35 @@ type GetWorkerBuildIdCompatibilityOptions struct {
 	MaxSets   int
 }
 
+type GetWorkerTaskReachabilityOptions struct {
+	// BuildIDs - The build IDs to query the reachability of. At least one build ID must be provided.
+	BuildIDs []string
+	// TaskQueues - The task queues with Build IDs defined on them that the request is
+	// concerned with.
+	// Optional: defaults to all task queues
+	TaskQueues []string
+	// Reachability - The reachability this request is concerned with.
+	// Optional: defaults to all types of reachability
+	Reachability TaskReachability
+}
+
+type WorkerTaskReachability struct {
+	// Reachable - map of build IDs and their reachability information
+	// May contain an entry with [UnversionedBuildID] for an unversioned worker
+	Reachable map[string]*BuildIDReachability
+}
+
+type BuildIDReachability struct {
+	// TaskQueueReachable map of task queues and their reachability information
+	TaskQueueReachable map[string]*TaskQueueReachability
+}
+
+type TaskQueueReachability struct {
+	// Reachability for a worker in a single task queue.
+	// If Reachability is empty, this worker is considered unreachable in this task queue.
+	Reachability []TaskReachability
```

Comment:
```suggestion
	TaskQueueReachability []TaskReachability
```

Seems to match others